src/main/java/io/stargate/sgv2/jsonapi/api/request/DataApiRequestInfo.java

```diff
@@ -54,4 +57,39 @@ public Optional<String> getCassandraToken() {
   public EmbeddingCredentials getEmbeddingCredentials() {
     return this.embeddingCredentials;
   }
+
+  public HttpHeaderAccess getHttpHeaders() {
+    return this.httpHeaders;
+  }
+
+  /**
+   * Simple wrapper around internal HTTP header container, providing safe(r) access to typed header
+   * values. Minimal API, currently mainly used for feature flags.
+   */
+  public static class HttpHeaderAccess {
+    private final io.vertx.core.MultiMap headers;
+
+    public HttpHeaderAccess(io.vertx.core.MultiMap headers) {
+      this.headers = headers;
+    }
+
+    /**
+     * Accessor for getting value of given header (case-insensitive), as {@code Boolean} if (and
```

Comment:
Yes, is due to vertx http server impl that does it for HTTP standard reason.

What I am thinking is that we only document use of lower-case header name for usage, and leave ability for case-insensitivity as impl detail: this since we are not exposing features to actual end users. If case-insensitivity wasn't provided I wouldn't bother actually adding it here.

Reply:
Hmmh. While I'd prefer all lower-case, looks like general best practice seems to be mixed case.

Reply:
I am on the edge of my seat here, seems to be ??? seems to be...

Reply:
hmmmh. Not sure why your GUI showed "seems to be" without second line ("mixed case"). Or is that secret line just shown to me? :-o
(I saw it missing when you screen shared)